spec/controllers/api/v1/base_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+
+RSpec.describe Api::V1::BaseController, type: :controller do
+  # assert token and options get parsed correctly
+  controller do
+    def index
+      render json: {message: "Successfully autenticated"}
+    end
+
+    def authenticated_user!
+      super
+    end
+  end
+
+  # assert authenticate_user! works
+  describe "GET #index" do
+    let(:user) { create(:volunteer) }
+    it "returns http success" do
```

Comment:
Remove comment and modify spec description:

```suggestion
    it "returns http success when valid credentials" do
```

Reply:
done